store/tikv/mock-tikv/executor.go

```diff
@@ -268,23 +289,22 @@ func (e *selectionExec) Next() (handle int64, value [][]byte, err error) {
 			return 0, nil, nil
 		}
 
-		err = e.eval.SetRowValue(handle, value, e.colIDs)
+		err = e.evalCtx.decodeRelatedColumnVals(e.relatedColOffsets, handle, value, e.row)
 		if err != nil {
 			return 0, nil, errors.Trace(err)
 		}
-		// TODO: Now the conditions length is 1.
-		result, err := e.eval.Eval(e.Conditions[0])
-		if err != nil {
-			return 0, nil, errors.Trace(err)
-		}
-		if result.IsNull() {
-			continue
-		}
-		boolResult, err := result.ToBool(e.eval.StatementCtx)
-		if err != nil {
-			return 0, nil, errors.Trace(err)
+		allMatch := true
```

Comment:
Make `evalBool` accepts expression would be simpler, we can then remove `allMatch` variable.